modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/AncillaryFileManager.java

```diff
@@ -515,29 +524,35 @@ public void resetSliceManager() throws IOException {
 
     /** Retrieve basic indexer properties by scanning the indexer XML instance. */
     protected void initIndexer() throws JAXBException {
-        if (indexerFile.exists() && indexerFile.canRead()) {
+        Indexer cachedIndexer = INDEXER_CACHE.get(indexerFile.getAbsolutePath());
+        if (cachedIndexer != null) {
+            indexer = cachedIndexer;
+        } else if (indexerFile.exists() && indexerFile.canRead()) {
             Unmarshaller unmarshaller = CONTEXT.createUnmarshaller();
             if (unmarshaller != null) {
                 indexer = (Indexer) unmarshaller.unmarshal(indexerFile);
-                // indexed information about dimensions that supports multiple bands
-                initMultipleBandsDimensionsInfo(indexer);
-                // Parsing schemas
-                final SchemasType schemas = indexer.getSchemas();
-                Map<String, String> schemaMapping = new HashMap<>();
-                if (schemas != null) {
-                    // Map schema names to schema attributes string
-                    List<SchemaType> schemaElements = schemas.getSchema();
-                    for (SchemaType schemaElement : schemaElements) {
-                        schemaMapping.put(schemaElement.getName(), schemaElement.getAttributes());
-                    }
+                INDEXER_CACHE.put(indexerFile.getAbsolutePath(), indexer);
+            }
+        }
+        if (indexer != null) {
```

Comment:
In the previous code the indexer was assumed to be there, in case the unmarshaller could be created.
Besides that... what happens if an unmarshaller or an index cannot be created? Should'nt an exception be thrown anyways?

Reply:
Added exceptions on those cases